blender/LilySurfaceScraper/Scrapers/AbstractScraper.py

```diff
@@ -102,61 +112,61 @@ def getTextureDirectory(self, material_name):
             texture_dir = texture_dir[2:]
         if not os.path.isabs(texture_dir):
             texture_dir = os.path.realpath(os.path.join(self.texture_root, texture_dir))
-        name_path = material_name.replace('/',os.path.sep)
+        name_path = material_name.replace('/', os.path.sep)
         dirpath = os.path.join(texture_dir, name_path)
         if not os.path.isdir(dirpath):
             os.makedirs(dirpath)
         return dirpath
 
-    def fetchImage(self, url, material_name, map_name, force_ext=False):
-        """Utility helper for download textures"""
-        root = self.getTextureDirectory(material_name)
-        if not force_ext:
-            ext = os.path.splitext(url)[1]
-            map_name = map_name + ext
-        path = os.path.join(root, map_name)
-        if os.path.isfile(path):
-            print("Using cached {}.".format(url))
-        else:
-            print("Downloading {}...".format(url))
-            headers = {"User-Agent":"Mozilla/5.0"}  # fake user agent
+    def downloadFunc(self, url, path):
+        def func():
```

Comment:
I think it would make more sense to have the func closure take the path as argument, otherwise there will always be some redundancy when calling `saveFile(path, some_callback_that_saves_to_path)`.